modules/@apostrophecms/modal/ui/apos/mixins/AposModalTabsMixin.js

```diff
@@ -60,14 +63,16 @@ export default {
 
   watch: {
     tabs() {
-      if ((!this.currentTab) || (!this.tabs.find(tab => tab.name === this.currentTab))) {
-        this.currentTab = this.tabs[0] && this.tabs[0].name;
+      if ((!this.currentTab) || (!this.tabs.find(tab => tab.isVisible === true && tab.name === this.currentTab))) {
+        const firstVisibleTab = this.tabs.find(tab => tab.isVisible === true);
+        this.currentTab = firstVisibleTab && firstVisibleTab.name;
```

Comment:
Slightly modified for readability (indentation + uses some instead of find):
```javascript
    tabs() {
      if (
        (!this.currentTab) ||
        (!this.tabs.some(tab => tab.isVisible === true && tab.name === this.currentTab))
      ) {
        const { name: firstVisibleTabName } = this.tabs.find(tab => tab.isVisible === true) || {};
        this.currentTab = firstVisibleTabName;
      }
    }
```
Since the code inside the if is redundant, it could be moved in a specific method.